ign_ros2_control/src/ign_system.cpp

```diff
@@ -369,6 +526,41 @@ bool IgnitionSystem::initSim(
     }
   }
 
+  rclcpp::Parameter joint_names_parameter("joints", joint_names);
+  if (!this->param_node_->has_parameter("joints")) {
+    this->param_node_->set_parameter(joint_names_parameter);
+  }
+  for (const auto & p : param_vec) {
+    if (!this->param_node_->has_parameter(p.get_name())) {
+      this->param_node_->set_parameter(p);
+    }
+  }
+
+  spin_thread_ = std::thread(
+    [this]() {
+      exec_ = std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+      exec_->add_node(this->param_node_);
+
+      while (rclcpp::ok() && !stop_spin_) {
+        exec_->spin_once();
+      }
+      exec_->remove_node(this->param_node_);
+      exec_.reset();
+    });
```

Comment:
With the way this is written, the thread will stop when the system is deactivated and `stop_spin` will remain true. Does this design mean it is not possible to deactivate and then later reactivate the system?

If this is not meant to be supported, then ignore this comment.

Reply:
It is not within the scope of this PR. In best case we should be able to deactivate and activate hardware interfaces in runtime. But let's but this for future work.
Nevertheless, good catch!